autorest/azure/token.go

```diff
@@ -70,29 +77,72 @@ func (t *Token) WithAuthorization() autorest.PrepareDecorator {
 type ServicePrincipalToken struct {
 	Token
 
+	secret        ServicePrincipalSecret
 	clientID      string
-	clientSecret  string
-	resource      string
 	tenantID      string
+	resource      string
 	autoRefresh   bool
 	refreshWithin time.Duration
 	sender        autorest.Sender
 }
 
-// NewServicePrincipalToken creates a ServicePrincipalToken from the supplied Service Principal
-// credentials scoped to the named resource.
-func NewServicePrincipalToken(id string, secret string, tenantID string, resource string) (*ServicePrincipalToken, error) {
+// ServicePrincipalSecret is an interface that allows various secret mechanism to fill the form
+// that is submitted when acquiring an oAuth token
+type ServicePrincipalSecret interface {
+	SetAuthValues(spt *ServicePrincipalToken, values *url.Values) error
+}
+
+// ServicePrincipalTokenSecret implements ServicePrincipalSecret for client_secret type authorization
+type ServicePrincipalTokenSecret struct {
+	ClientSecret string
+}
+
+// ServicePrincipalCertSecret implements ServicePrincipalSecret for cert auth with signed JWTs
```

Comment:
I believe so. I am a bit odd on this. I react strongly against the "hacker" culture at this point. I have had to maintain code with poor variable names and no comments. Those are not happy times. Hence, I push for good variable names, fully spelled out (without becoming the horrifically long Java-esque style). Ok?